ui/app/adapters/job.js

```diff
@@ -1,11 +1,35 @@
 import Ember from 'ember';
 import ApplicationAdapter from './application';
 
-const { RSVP } = Ember;
+const { RSVP, inject } = Ember;
 
 export default ApplicationAdapter.extend({
+  system: inject.service(),
+
   shouldReloadAll: () => true,
 
+  buildQuery(snapshot) {
+    const namespace = this.get('system.activeNamespace');
+
+    // SnapshotRecordArray isn't exported, so the best we can do is duck-type.
+    const isSnapshotRecordArray = snapshot && snapshot._recordArray;
+    if ((!snapshot || isSnapshotRecordArray) && namespace) {
+      return {
+        namespace: namespace.get('name'),
+      };
+    }
+  },
+
+  findAll() {
+    const namespace = this.get('system.activeNamespace');
+    return this._super(...arguments).then(data => {
+      data.forEach(job => {
+        job.NamespaceID = namespace && namespace.get('id');
```

Comment:
Would it be better to put this in the serializer in a normalization hook? Not sure it'd save any iteration, but it'd be pre ember object creation.

Reply:
The reason I put it in the adapter is it smooths over a difference between the `jobs` endpoint and the `job/:id` endpoint.

I also initially thought I was going to implement it using the URL, since the reasoning behind the api is Namespace doesn't need to be in the response since it's already a query param on the request. Since I'm using the system state rather than the URL, I could make it work in the serializer, but I don't see a particular reason to, since this is also pre ember object creation.